### PR TITLE
SAIC-173 Create cinder volumes from generate_load

### PR DESCRIPTION
**Pull request by plomakin**
* Added volumes generating.
* Added snapshot generating.
* Snapshot attaching to vm.
* Configuration's template changes added.
* SAIC-183: Added resource validation for cinder objects.
* SAIC-206: Optimized config's parameters passing to functions.
* Fixed networking cleanup.
* Added roles generating.
* Optimized cleaning scenario output.

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by oleksii-shyman** [comment]
Code is ok - But the point is - we have a lot of functions already defined in resources, probably - we have to think about how to reuse that code.
Anyways +1 from my side

**Comment by plomakin** [comment]
@oleksii-shyman one of the points for the test automation - we can't use developer's code... we actually test it.

**Comment by miarmak** [comment]
+1

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Build finished. Test PASSed.

**Comment by Wasapl** [comment]
Build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by miarmak** [comment]
Looks nice. +1

File: devlab/tests/config.py [hunk 1]
```diff
@@ -34,35 +39,46 @@
...
           },
-         {'name': 'sg12', 'description': 'Blah blah group2'}]
+         {'name': 'sg12', 'description': 'Blah blah group2'}],
+     'cinder_volumes': [
+         {'name': 'tn1_volume1', 'size': 1, 'server_to_attach': 'tn1server1',
+          'device': '/dev/vdb'},
+         {'name': 'tn1_volume2', 'size': 1}
+     ],
+     'cinder_snapshots': [
+         # {'name': 'tn1snapsh', 'volume_id': 'tn1_volume2'}
```
Comment (roman-verchikov) [inline devlab/tests/config.py]: Let's put a comment describing why snapshot creation is disabled in our test scenario

File: devlab/tests/config.py [hunk 2]
```diff
@@ -78,6 +94,8 @@
 # Flavors to create/delete
 flavors = [
     {'name': 'flavorname1', 'disk': '10', 'ram': '32', 'vcpus': '1'},
+    # {'name': 'flavorname3', 'disk': '10', 'ram': '32', 'vcpus': '1',
+    #  'is_public': False},
```
Comment (roman-verchikov) [inline devlab/tests/config.py]: Why is this commented?

File: devlab/tests/config.py [hunk 3]
```diff
@@ -114,12 +132,25 @@
...
+# Cinder images to create/delete
+cinder_volumes = [
+    {'name': 'cinder_volume1', 'size': 1},
+    {'name': 'cinder_volume2', 'size': 1,
+     'server_to_attach': 'server2', 'device': '/dev/vdb'}
+]
+
+# Cinder snapshots to create/delete
+cinder_snapshots = [
+    # {'display_name': 'snapsh1', 'volume_id': 'cinder_volume1'}
```
Comment (roman-verchikov) [inline devlab/tests/config.py]: Please put a comment describing why snapshot testing is disabled

**Comment by roman-verchikov** [comment]
Looks good other than couple commented lines in config which are somewhat confusing and require description

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by roman-verchikov** [comment]
+1

**Pull request merged**